hs_app_timeseries/models.py

```diff
@@ -1495,333 +1740,310 @@ def _update_timeseriesresultvalues_table_insert(self, con, cur, temp_csv_file, r
-        # insert record to Sites table - first delete any existing records
-        # this function used only in case of writing data to a blank database (case of CSV upload)
+        cur.execute("DELETE FROM People")
+        con.commit()
+        insert_sql = "INSERT INTO People (PersonID, PersonFirstName, " \
+                     "PersonMiddleName, PersonLastName) VALUES(?,?,?,?)"
+        people_data = []
+        for index, person in enumerate(list(self.creators.all()) +
+                                       list(self.contributors.all())):
+            person_id = index + 1
+            name_parts = person.name.split()
+            first_name = name_parts[0]
+            mid_name = ''
+            last_name = ''
+            if len(name_parts) > 2:
+                mid_name = name_parts[1]
+                last_name = name_parts[2]
+            elif len(name_parts) == 2:
+                last_name = name_parts[1]
+            cur.execute(insert_sql, (person_id, first_name, mid_name, last_name), )
+            is_creator = isinstance(person, Creator)
+            people_data.append({'person_id': person_id,
+                                'organization': person.organization,
+                                'email': person.email if person.email else '',
+                                'phone': person.phone,
+                                'address': person.address,
+                                'is_creator': is_creator,
+                                'object_id': person.id})
+        return people_data
 
-        cur.execute("DELETE FROM Sites")
+    def update_organizations_table_insert(self, con, cur):
+        # insert record to Organizations table - first delete any existing records
+        # used for updating a sqlite file that is blank (case of CSV upload)
+
+        cur.execute("DELETE FROM Organizations")
         con.commit()
-        insert_sql = "INSERT INTO Sites (SamplingFeatureID, SiteTypeCV, Latitude, " \
-                     "Longitude, SpatialReferenceID) VALUES(?,?,?,?,?)"
-        for index, site in enumerate(self.sites):
-            sampling_feature_id = index + 1
-            spatial_ref_id = 1
-            cur.execute(insert_sql, (sampling_feature_id, site.site_type, site.latitude,
-                                     site.longitude, spatial_ref_id), )
+        organizations = []
+        org_id = 1
+        insert_sql = "INSERT INTO Organizations (OrganizationID, OrganizationTypeCV, " \
+                     "OrganizationCode, OrganizationName) VALUES(?,?,?,?)"
+        for person in (list(self.creators.all()) + list(self.contributors.all())):
+            organization_name = person.organization if person.organization else 'Unknown'
+            if organization_name not in organizations:
+                organizations.append(organization_name)
+                cur.execute(insert_sql, (org_id, 'Unknown', str(org_id),
+                                         organization_name), )
```

Comment:
Not sure why always insert ```Unknown``` for OrganizationtypeCV, and str(org_id) for OrganizationCode? If OrganizationCode is always str(org_id), OrganizationID field can be directly used, why need another field?

Reply:
Not sure I understand your question. Since we don't have a value for organizationtypeCV we are entering 'Unknown' .

Reply:
What I mean is ```OrganizationCode``` uses the same org_id as ```OrganizationID```, so I am wondering why we need ```OrganizationCode``` field. Now that I am thinking about it, perhaps ```OrganizationCode``` is editable and can be made different from ```OrganizationID```, and it is just set to org_id initially as a default value?

Reply:
The field ```OrganizationCode``` is part of the ODM2 database schema for that table. So I am not creating an additional field here. I am just providing a value for the existing fields of the table.